lib/tasks_private/spec.rake

```diff
@@ -18,8 +18,31 @@ namespace :spec do
   desc "Prepare all specs"
   task :setup => [:initialize, :setup_db]
 
+  desc <<~DESC
+    Generate table list for schema specs
+
+    Generates a list of tables from the `spec/dummy/db/schema.rb` to be used in
+    the `schema_dumper_spec.rb`.
+  DESC
+  task :table_list => "spec:migrations:complete_up" do
+    table_list = File.open("spec/support/table_list.txt", "w")
+    File.open("spec/dummy/db/schema.rb").each_line do |line|
+      next unless line =~ /^\s*create_table\s+"([^"]+)"/
+
+      table_list.puts $1
+    end
+  ensure
+    table_list.close
+  end
+
   desc "Run all non-migration specs"
   RSpec::Core::RakeTask.new(:non_migration => :initialize) do |t|
     t.pattern = FileList["spec/**/*_spec.rb"].exclude("spec/migrations/**/*_spec.rb")
+                                             .exclude("spec/schema/**/*_spec.rb")
+  end
+
+  desc "Run schema specs"
+  RSpec::Core::RakeTask.new(:schema => [:initialize, "spec:migrations:complete_up"]) do |t|
```

Comment:
I don't think we need to depend on `spec:migrations:complete_up`.  It is ok to assume that the database will already be fully upgraded, similar to how we do for `spec:non-migration` and even `spec:migration`.  All we need to ensure is that the database is left in the up state when this spec completes.